toolchain/check/testdata/generic/complete_type.carbon

```diff
@@ -0,0 +1,259 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/generic/complete_type.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/generic/complete_type.carbon
+
+// --- fail_incomplete_in_class.carbon
+
+library "[[@TEST_NAME]]";
+
+class A(T:! type) {
+  // CHECK:STDERR: fail_incomplete_in_class.carbon:[[@LINE+3]]:10: error: `T` evaluates to incomplete type `B` [IncompleteTypeInMonomorphization]
+  // CHECK:STDERR:   var v: T;
+  // CHECK:STDERR:          ^
+  var v: T;
+}
+
+// CHECK:STDERR: fail_incomplete_in_class.carbon:[[@LINE+4]]:1: note: class was forward declared here [ClassForwardDeclaredHere]
+// CHECK:STDERR: class B;
+// CHECK:STDERR: ^~~~~~~~
+// CHECK:STDERR:
+class B;
```

Comment:
I'm not sure if this ordering is deliberate, but if you forward declare `B` before `A`, it would make the diagnostic contiguous and possibly easier to read in the test.

Reply:
Nice, thanks, done.